src/distributions/adopt/installer.ts

```diff
@@ -8,12 +8,14 @@ import semver from 'semver';
 import { JavaBase } from '../base-installer';
 import { IAdoptAvailableVersions } from './models';
 import { JavaInstallerOptions, JavaDownloadRelease, JavaInstallerResults } from '../base-models';
-import { MACOS_JAVA_CONTENT_POSTFIX } from '../../constants';
 import { extractJdkFile, getDownloadArchiveExtension, isVersionSatisfies } from '../../util';
 
-export class AdoptDistribution extends JavaBase {
-  constructor(installerOptions: JavaInstallerOptions) {
-    super('Adopt', installerOptions);
+export abstract class AdoptDistribution extends JavaBase {
+  private impl: string;
+
+  constructor(impl: string, installerOptions: JavaInstallerOptions) {
```

Comment:
Considering the fact that both Hotspot and OpenJ9 have almost identical logic, I think we don't need to create the separate classes and inheritance here.
It is enough to add `jvm-impl` parameter to `AdoptDistribution` constructor.

So in `distribution-factory.ts` we will create an object like
```
case JavaDistribution.JdkFile:
      return new LocalDistribution(installerOptions, jdkFile);
    case JavaDistribution.Adopt:
    case JavaDistribution.AdoptHotspot:
      return new AdoptDistribution(installerOptions, 'Hotspot');
    case JavaDistribution.AdoptOpenJ9:
      return new AdoptDistribution(installerOptions, 'OpenJ9');
    case JavaDistribution.Zulu:
      return new ZuluDistribution(installerOptions);
```

It will simplify code and we won't need to duplicate tests